src/text-field.js

```diff
@@ -30,81 +27,48 @@ class Textfield extends React.Component {
   }
 
   render(){
-
     const {
-      className,
-      type,
-      label,
-      floatingLabel,
-      error,
+      children,
       expandable,
-      button,
       icon,
-      expandableHolder
+      floating,
+      className
     } = this.props;
 
-    let {
-      id
-    } = this.props;
-
-    const outerClasses = classnames(outerBaseClasses, {
-      'mdl-textfield--floating-label': floatingLabel,
-      'mdl-textfield--expandable': expandable,
-      'mdl-button': button,
-      'mdl-js-button': button,
-      'mdl-button--icon': icon,
-      'mdl-textfield__expandable-holder': expandableHolder
-    }, className);
-
-    const inputClasses = classnames(inputBaseClasses);
+    let {id} = this.props;
 
-    const labelClasses = classnames(labelBaseClasses);
-
-    let textField;
-    let labelField;
-    let errorField;
-
-    if (!id) {
-      id = '_' + Math.random().toString(36).slice(2);
+    if(!id){
+      id = this._autoId;
     }
 
-    if (type === 'textarea') {
-      textField = (<textarea className={inputClasses} id={id} />);
-    } else {
-      textField = (<input className={inputClasses} type={type} id={id} />);
-    }
-
-    if (label) {
-      labelField = (<label className={labelClasses} htmlFor={id}>{label}</label>);
-    }
+    const inputChildren = React.Children.map(this.props.children, function(child) {
+      return React.cloneElement(child, {
```

Comment:
Due to the fact that that the key element with input and textarea is "name" and "value" keys with html forms.  Also if you choose an onclick or onchange method, you're going to be using a ref method to refer to the input box.  I went ahead and added the check for TextFieldLabel, and kept it strict to that.